src/util/invariant.h

```diff
@@ -227,13 +227,23 @@ inline void invariant_violated_string(
 // The condition should only contain (unmodified) arguments to the method.
 // "The design of the system means that the arguments to this method
 //  will always meet this condition".
+//
+// The condition asserts properties that the function expects, including,
+// member variables of the object, in case they are implicit input to
+// the function. It is also supposed to signal that even though an object
+// is in a valid state, a particular function may not be operating on
+// objects with the particular properties this one enjoys.
```

Comment:
enjoys?

Reply:
yes, this is why I wanted a review. Maybe an Englishman can chime in? I don't know what a proper expression for that would be, but I don't think that what's there is unreasonable.

Reply:
Instead of adding the first part of this description, can we rephrase the existing one above? Maybe something like `The condition should only contain (unmodified) inputs to the method. Inputs include arguments passed to the method, as well as member variables that the
method may read.`